peer/brontide.go

```diff
@@ -629,7 +629,30 @@ func (p *Brontide) Start() error {
 	go p.writeHandler()
 	go p.readHandler()
 	go p.channelManager()
-	go p.pingHandler()
+
+	var (
+		lastBlockHeader           *wire.BlockHeader
+		lastSerializedBlockHeader [wire.MaxBlockHeaderPayload]byte
+	)
+	newPingPayload := func() []byte {
+		// Each time a new block comes in, we'll copy the raw header
+		// contents over to our ping payload declared above. Over time,
+		// we'll use this to disseminate the latest block header
+		// between all our peers, which can later be used to
+		// cross-check our own view of the network to mitigate various
+		// types of eclipse attacks.
+		header := p.cfg.CurrentChainStateView.BestBlockHeader()
+		if header != lastBlockHeader {
+			buf := bytes.NewBuffer(lastSerializedBlockHeader[0:0])
+			err := header.Serialize(buf)
+			if err != nil {
+				lastBlockHeader = header
```

Comment:
Log why we can't serialize? Tho should never happen in practice...

Reply:
Yeah this is a classic case of, "trust me, dear compiler, I assure you this scenario is impossible". I can definitely log something to the effect that it should be impossible and that the log message showing up implies an implementation bug.